### ISSUE TITLE
The caret is always set to the far right when deleting or inserting numbers when using the `$` currency sign

### ISSUE THREAD

**Issue by jroozee** [autoNumeric/autoNumeric#684]
### Current behavior
If user enters "$123.45" then places cursor to left of decimal  (e.g. "$123|.45"), presses Backspace to delete the 3 or types a new number,  the cursor moves all the way to the right. If user presses backspace again, the 5 is then deleted.

### Expected behavior
If user enters $123.45 then places cursor to left of decimal (e.g. "$123|.45"), presses Backspace to delete the 3 or types a new number, the cursor should remain where it is. 

### Steps to reproduce the problem
1. Use autoNumeric version 4.6.0
2. In the browser Chrome version 81 and 83.
3. On the Windows operating system
4. Then instantiate the autoNumeric object with the following options:
```js
CurrencySymbol: "$",
    decimalCharacterAlternative: ",",
    decimalPlacesRawValue: 2,
    decimalPlacesShownOnBlur: 2,
    decimalPlacesShownOnFocus: 2,
    emptyInputBehavior: "always",
    historySize: 5,
    maximumValue: "999999",
    minimumValue: "0",
    negativePositiveSignPlacement: "p",
    onInvalidPaste: "clamp",
    unformatOnSubmit: true,
    emptyInputBehavior: "zero",
    modifyValueOnWheel: false
```

### Link to live example (ie. [Codepen](http://codepen.io/pen/))
https://www.quantumpayments.com/sandbox/Practice/autoNumericExample.html

**Comment by jroozee** [comment]
To clarify, this issue is different and unrelated to #685

**Comment by AlexandreBonneau** [comment]
I can reproduce the bug using that [code sample](https://codepen.io/AnotherLinuxUser/pen/abdzWLd?editors=1010).
So far, when using any other currency sign than `$`, it works as expected.
I suspect a faulty regex is at play here.

**Comment by Diamentix** [comment]
> I can reproduce the bug using that [code sample](https://codepen.io/AnotherLinuxUser/pen/abdzWLd?editors=1010).
> So far, when using any other currency sign than `$`, it works as expected.
> I suspect a faulty regex is at play here.

I've found another case, where this seems to occur - right now whenever I have a negative value (I used the code sample You provided) - whether it's `$` or `€` - the same issue occurs when I delete a digit. 

### Steps to reproduce:
1. Use linked code sample
2. Set the config to the one with `€` (comment the `$` one)
3. Type a value
4. Add a `-` into it
5. The caret moves to the end of the value
Additionally:
6. Move caret between the digits (regardless of integer or decimals)
7. Remove any digit (either backspace or delete)
8. The caret once again moves to the end of the value

However an important factor is that it only occurs in select few negative value configs, namely:
`negativePositiveSignPlacement: "l"` or `negativePositiveSignPlacement: "p"`
The suffix and right values seem to work properly. 
So basically either for the `$` itself or any currency symbol with a `-` before it - it seems to reposition the caret to the end of the input value.

**Comment by jasonlfunk** [comment]
Yes, this is definitely related to using a dollar sign as the currencySymbol. A silly workaround is to use a different unicode dollar sign, such as: https://www.compart.com/en/unicode/U+FF04

**Comment by Blackbaud-TrevorBurch** [comment]
@AlexandreBonneau - I see that a fix for this was merged into the codebase on July 7. However, it does not seem that a release has been cut with this fix. Would you happen to have an ETA for when a release might happen for this fix? Thanks for any info! 😄
